Add Telegram notification on PR to review and merge on main

masch opened the pull request:
Using GH actions, add support to send Telegram notifications to a group when:

1.  PR is ready to review.
2.  PR was merged on main.

In order to make it work, the following tasks must be done:
- Create the following secrets on the project: 
  - TELEGRAM_TO: Contains the unique identifier for the chat group. 
  - TELEGRAM_TOKEN: TG authorization token.

  More information about the values: https://github.com/appleboy/telegram-action#secrets

- Add a TG bot on the chat to post the message.

- Merge the current PR configuration.

More information about the GH action used: https://github.com/marketplace/actions/telegram-notify

masch committed:
Add TG notif on PR to review and merge on main

Using GH actions, add support to send Telegram notifications on a
 group when a PR is ready to review or it was merged on main.

In order to make it work, the following tasks must be done:
- Create the following secrets on the project:
TELEGRAM_TO: Contains the unique identifier for the chat group.
TELEGRAM_TOKEN: TG authorization token.

More information about the values:
https://github.com/appleboy/telegram-action#secrets

- Add a TG bot on the chat to post the message.

- Merge the current PR configuration.

More information about the GH action used:
https://github.com/marketplace/actions/telegram-notify

ptaq-eth commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,37 @@
+    steps:
+      - name: Send telegram message on pull request to review
+        if: github.event.pull_request.merged == false && github.event.action != 'closed'
+        uses: appleboy/telegram-action@master
```
How about freezing version with commit hash? This approach results in less surprises or damages by malware.

masch replied:
Sorry, I don't follow your suggestion. Are you suggesting changing the preconditions to sending the notification? Are you suggesting using the commit hash to trigger it? If that the case, anyone how pushes a commit on any PR will also trigger the notification.
Maybe I misunderstood you, could you extend on your suggestion?

ptaq-eth replied:
I was suggesting using `@<commit-hash>`  instead of `@master` (dynamic tip of branch) when referring to action repository. Perhaps it's not so critical on OSS projects, but always a potential vector.

masch replied:
You are right. I changed it to use the commit hash related to the current latest tag for v.0.1.1.

ptaq-eth commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,37 @@
+        if: github.event.pull_request.merged == true
+        uses: appleboy/telegram-action@master
+        with:
+          to: ${{ secrets.TELEGRAM_TO }}
```
@santisiri could you help with this? I think only have permissions to set these secrets.

masch committed:
Use commit hash on TG GH action plugin version

In order to approach a less surprises or damages by malware,
a commit hash is used instead of the master branch on the
appleboy/telegram-action Github action.
The commit hashed used in the one associated on the v0.1.1:
2e9996f96e095a537aa4442da4af41ca7b594fba
